google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/GoogleComputeEngineCredentialSupplierFromJson.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.googlecomputeengine;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.jclouds.domain.Credentials;
+
+import com.google.common.base.Supplier;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/*
+ * Provides an easy way to pass in credentials using the json-key format.
+ * Just provide the path to the .json file and this extracts and sets identity
+ *  and credentials from the json.
+ */
+public class GoogleComputeEngineCredentialSupplierFromJson implements Supplier<Credentials>{
+
+   private Credentials creds;
+
+   private GoogleComputeEngineCredentialSupplierFromJson(String jsonString){
+      creds = parseJsonKeyString(jsonString);
+   }
+
+   /**
+    * Function for parsing JSON Key file downloaded from GCP developers console.
+    *
+    * @param pathToJsonFile.
+    * @return Credentials object with Credentials.identity and Credentials.credential correctly set.
+    */
+   public static Credentials parseJsonKeyString(String jsonString) {
+      // Parse JsonFile to extract Service Account and PrivateKey.
+      final JsonObject json = new JsonParser().parse(jsonString).getAsJsonObject();
+      String client_email = json.get("client_email").toString().replace("\"", "");
+      String private_key = StringEscapeUtils.unescapeJava(json.get("private_key").toString().replace("\"", ""));
```

Comment:
Without the unescapeJava it was was incorrectly treating \n as two separate characters instead of newlines. The client email doesn't have any \n characters so it is a non-issue.

I found a simpler way to fix the newline character.
`json.get("private_key").toString().replace("\"", "").replace("\\n", "\n");`